daemon/api.go

```diff
@@ -420,6 +405,72 @@ func shouldSearchStore(r *http.Request) bool {
 	return false
 }
 
+func storeUpdates(c *Command, r *http.Request, user *auth.UserState) Response {
+	route := c.d.router.Get(snapCmd.Path)
+	if route == nil {
+		return InternalError("cannot find route for snaps")
+	}
+
+	found, err := allLocalSnapInfos(c.d.overlord.State())
+	if err != nil {
+		return InternalError("cannot list local snaps: %v", err)
+	}
+
+	localSnapsInfo := make([]*store.CurrentSnap, len(found))
+	localSnapMap := map[string]*snap.Info{}
+	for i, sn := range found {
+		localSnapsInfo[i] = &store.CurrentSnap{
+			// the desired channel (not sn.info.Channel!)
+			Channel: sn.snapst.Channel,
+
+			SnapID:      sn.info.SnapID,
+			Revision:    sn.info.Revision,
+			Epoch:       sn.info.Epoch,
+			Confinement: sn.info.Confinement,
+		}
+		localSnapMap[sn.info.Name()] = sn.info
+	}
+
+	// the store gives us everything, we need to client side filter
+	// for the updates
+	store := newRemoteRepo()
+	allUpdates, err := store.Updates(localSnapsInfo, user.Authenticator())
```

Comment:
It'd be nice if we could attempt to align terminology after the "refresh" concept, at least inside snapd itself (I reckon we cannot easily change store-side naming). So, this might be ListRefresh or similar, for example.